Java 17

tsatam opened the pull request:
# WIP DONOTMERGE

# This branch contains all the changes from the Angular deprecation branch. At the very least, this branch should not be merged into main until after the Angular deprecation branch is merged in. 

## Overview

Update RetroQuest to Java 17 and use new language features

### Notes
We already take advantage of [Gradle Toolchains](https://docs.gradle.org/current/userguide/toolchains.html), so this should have no impact to any users who would like to run RetroQuest themselves. This will not create a requirement that users must have a Java 17 JDK installed on their machine.

## Testing Instructions
- Run existing tests and ensure they still pass (there will be no functionality changes as a part of this pull request)

LowBudgetMan commented on api/src/test/java/com/ford/labs/retroquest/api/ThoughtApiTest.java:
```diff
@@ -138,27 +144,28 @@ public void should_not_move_thought_column_unauthorized() throws Exception {
     public void should_move_thought_column() throws Exception {
-        ColumnTitle savedColumnTitle = columnTitleRepository.save(new ColumnTitle(null, "cheeseburgers", "CheeseBurgers", teamId));
+        ColumnTitle savedColumnTitle = columnTitleRepository.save(new ColumnTitle(null, "cheeseburgers", "CheeseBurgers",
+            teamId));
```
Is this supposed to be orphaned on a new line?

LowBudgetMan commented on api/src/test/java/com/ford/labs/retroquest/api/ThoughtApiTest.java:
```diff
@@ -171,31 +178,32 @@ public void should_move_thought_column() throws Exception {
-        List<Thought> actualThoughts = objectMapper.readValue(result.getResponse().getContentAsByteArray(), new TypeReference<>() {
-        });
+        List<Thought> actualThoughts = objectMapper.readValue(result.getResponse().getContentAsByteArray(),
+            new TypeReference<>() {
```
Same with this one

tsatam commented:
Fixed formatting, updated the Actions pipelines, and updated the main README to not specify a specific Java version.

The Actions pipeline technically does not need to be updated (builds were still passing) since any JDK 8 or newer is sufficient to build the app since we use Gradle toolchains.

LowBudgetMan reviewed (approved):
lgtm
